persistent/picklecache.py

```diff
@@ -379,17 +463,14 @@ def _sweep(self, target, target_size_bytes=0):
 
                 value._p_deactivate()
                 if (self._persistent_deactivate_ran
-                    # Test-cases sneak in non-Persistent objects, sigh, so naturally
-                    # they don't cooperate (without this check a bunch of test_picklecache
-                    # breaks)
-                    or not isinstance(value, self._SWEEPABLE_TYPES)):
-                    to_eject.append((i, value))
+                        # Test-cases sneak in non-Persistent objects, sigh, so naturally
+                        # they don't cooperate (without this check a bunch of test_picklecache
+                        # breaks)
+                        or not isinstance(value, self._SWEEPABLE_TYPES)):
+                    ring.delete_node(node)
```

Comment:
Just double-checking: does `self.ring` support mutation during iteration?

Reply:
Happily now it does!

I took this opportunity to improve the semantics of the one-ond-only Ring implementation. It resulted in, i think, some nice simplifications.